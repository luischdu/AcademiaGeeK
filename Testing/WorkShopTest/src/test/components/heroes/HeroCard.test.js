import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { HeroCard } from '../../../components/heroes/HeroCard'

describe('Pruebas componente <HeroCard />', () => {
    test('Validar informacion Alter-Ego', () => {
        const id = 'marvel-thor'
        const superhero = 'Thor'
        const alter_ego = 'Thor Odinson'
        const first_appearance = 'Journey into Myster #83'
        const characters = 'Thor Odinson'
        
        const wrapper = shallow(
            <HeroCard
                id = {id}
                superhero = {superhero}
                alter_ego = {alter_ego}
                first_appearance = {first_appearance}
                characters = {characters}
            />
        )

        expect(wrapper).toMatchSnapshot()

        const textParrafo = wrapper.find('p').first().text().trim()
    
        expect(textParrafo).toBe('Thor Odinson')
    })
})