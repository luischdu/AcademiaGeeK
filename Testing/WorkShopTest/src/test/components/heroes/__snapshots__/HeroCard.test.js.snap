// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <HeroCard /> Validar informacion Alter-Ego 1`] = `
<div
  className="card ms-3 animate__animated animate__fadeIn"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="Thor"
        className="card-img"
        src="./assets/heroes/marvel-thor.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Thor
           
        </h5>
        <p
          className="card-text"
        >
           
          Thor Odinson
           
        </p>
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
             
            Journey into Myster #83
             
          </small>
        </p>
        <Link
          to="./hero/marvel-thor"
        >
          MÃ¡s...
        </Link>
      </div>
    </div>
  </div>
</div>
`;
